File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOGpsControlModel.swift
//
//  IDOGpsControlModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Control GPS event number
@objcMembers
public class IDOGpsControlModel: NSObject, IDOBaseModel {
    /// 1: Enable log 
    /// 2: Disable log 
    /// 3: Write AGPS data 
    /// 4: Erase AGPS data 
    /// 5: Write GPS firmware 
    public var type: Int
    /// 0: Invalid 
    /// 1: Command in progress 
    /// 2: Completed    
    public var status: Int
    /// 0 for success, non-zero for failure                            
    public var errorCode: Int
    
    enum CodingKeys: String, CodingKey {
        case type = "type"
        case status = "status"
        case errorCode = "error_code"
    }
    
    public init(type: Int,status: Int,errorCode: Int) {
        self.type = type
        self.status = status
        self.errorCode = errorCode
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOGpsControlModel.kt
//
//  IDOGpsControlModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Control GPS event number

open class IDOGpsControlModel(type: Int, status: Int, errorCode: Int) : IDOBaseModel {

    /// 1: Enable log 
    /// 2: Disable log 
    /// 3: Write AGPS data 
    /// 4: Erase AGPS data 
    /// 5: Write GPS firmware 
    @SerializedName("type")
    var type: Int = type

    /// 0: Invalid
    /// 1: Command in progress 
    /// 2: Completed    
    @SerializedName("status")
    var status: Int = status

    /// 0 for success, non-zero for failure
    @SerializedName("error_code")
    var errorCode: Int = errorCode


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    